src/plugins/matrixops/min_operation.cpp

```diff
@@ -0,0 +1,284 @@
+        "Returns the minimum of an array or minimum along an axis.")
+    };
+
+    ///////////////////////////////////////////////////////////////////////////
+    min_operation::min_operation(primitive_arguments_type&& operands,
+        std::string const& name, std::string const& codename)
+        : primitive_component_base(std::move(operands), name, codename)
+    {}
+
+    template <typename T>
+    primitive_argument_type min_operation::min0d(ir::node_data<T>&& arg,
+        hpx::util::optional<std::int64_t> axis) const
+    {
+        if (axis && axis.value() != 0 && axis.value() != -1)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "min_operation::min0d",
+                generate_error_message(
+                    "the min_operation primitive requires operand axis to be "
+                    "either 0 or -1 for scalar values."));
+        }
+
+        return primitive_argument_type{arg.scalar()};
+    }
+
+    template <typename T>
+    primitive_argument_type min_operation::min1d(ir::node_data<T>&& arg,
+        hpx::util::optional<std::int64_t> axis, bool keep_dims) const
+    {
+        if (axis && axis.value() != 0 && axis.value() != -1)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "min_operation::min1d",
+                generate_error_message(
+                    "the min_operation primitive requires operand axis to be "
+                    "either 0 or -1 for vectors."));
+        }
+        auto v = arg.vector();
+        T result = blaze::min BOOST_PREVENT_MACRO_SUBSTITUTION(v);
+
+        if (keep_dims)
+        {
+            return primitive_argument_type{blaze::DynamicVector<T>{result}};
+        }
+        return primitive_argument_type{ir::node_data<T>{result}};
+    }
+
+    template <typename T>
+    primitive_argument_type min_operation::min2d(ir::node_data<T>&& arg,
+        hpx::util::optional<std::int64_t> axis, bool keep_dims) const
+    {
+        if (axis)
+        {
+            switch (axis.value())
+            {
+            case -2:
+                HPX_FALLTHROUGH;
+            case 0:
+                return min2d_axis0(std::move(arg), keep_dims);
+
+            case -1:
+                HPX_FALLTHROUGH;
+            case 1:
+                return min2d_axis1(std::move(arg), keep_dims);
+
+            default:
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "min_operation::min2d",
+                    generate_error_message(
+                        "the min_operation primitive requires operand axis "
+                        "to be between -2 and 1 for matrices."));
+            }
+        }
+        return min2d_flat(std::move(arg), keep_dims);
+    }
+
+    template <typename T>
+    primitive_argument_type min_operation::min2d_flat(
+        ir::node_data<T>&& arg, bool keep_dims) const
+    {
+        auto m = arg.matrix();
+        T result = blaze::min BOOST_PREVENT_MACRO_SUBSTITUTION(m);
+
```

Comment:
It's easier to write this as: `(blaze::min)(m);` as this also prevents `min` being expanded as a macro.